projects/fsastorefrontlib/src/core/my-account/facade/consent.service.ts

```diff
@@ -23,11 +31,46 @@ export class ConsentService {
     );
   }
 
+  transferCartToSelectedOBOCustomer(
+    cart: FSCart,
+    currentUser: FSUser,
+    oboConsentCustomer: FSUser
+  ) {
+    this.store.dispatch(
+      new fromAction.TransferCart({
+        cart: cart,
+        consentHolder: currentUser,
+        oboCustomer: oboConsentCustomer,
+      })
+    );
+  }
+
+  isCartTransferAllowedForSeller(): Observable<boolean> {
+    return combineLatest([
+      this.userAccountFacade.get(),
+      this.selectedOBOCustomer$,
+    ]).pipe(
+      map(([seller, oboConsentCustomer]) => {
```

Comment:
Maybe you could use filter operator since you are doing filtering and write something like this:
`filter(([seller, oboConsentCustomer]) => !seller?.roles.includes(FSUserRole.SELLER) || oboConsentCustomer?.uid )`. That should do the same thing in more clean way.

Reply:
Since here we need this method to return either true or false, and filter is returning the whole object based on some condition, I don't think it's applicable here for code optimizing. Since after filter we would have to use map again afterwards to return true or false I would suggest we leave it like this. @bojanjevtic-spa maybe you have a better insight into this, since you wrote it?

Reply:
You are right, the output should be boolean, so i would stick with returning true/false based on some conditions